add definition link of HRESULT

ksyx opened the pull request.

ksyx committed:
doc: definition of HRESULT

PRMerger15 commented:
@ksyx : Thanks for your contribution! The author(s) have been notified to review your proposed change.

mcleanbyron committed:
Changed link target

Changed link target to HRESULT documentation in Windows specs instead of unrelated page in Office docs.

PRMerger18 commented:
@mcleanbyron : Thanks for your contribution! The author(s) have been notified to review your proposed change.
